plugins/building-hacks.cpp

```diff
@@ -20,47 +20,56 @@
 #include "df/tile_building_occ.h"
 #include "df/building_drawbuffer.h"
 #include "df/general_ref_creaturest.h" // needed for power information storage
+#include "df/building_def_workshopst.h"
 #include "modules/Buildings.h"
 
 #include <map>
-
+#include <limits>
 using namespace DFHack;
 using namespace df::enums;
 
 DFHACK_PLUGIN("building-hacks");
 REQUIRE_GLOBAL(world);
 
+constexpr uint32_t invalid_tile = std::numeric_limits<uint32_t>::max();
 struct graphic_tile //could do just 31x31 and be done, but it's nicer to have flexible imho.
 {
-    int16_t tile; //originally uint8_t but we need to indicate non-animated tiles
+    int16_t tile=-1; //originally uint8_t but we need to indicate non-animated tiles
     int8_t fore;
     int8_t back;
     int8_t bright;
+    //index of texpos
+    uint32_t graphics_tile = invalid_tile;
```

Comment:
the texpos vectors you will be writing these into are `int32_t`. why the type difference?